core/src/main/java/org/apache/iceberg/BaseRewriteFiles.java

```diff
@@ -101,7 +101,7 @@ public RewriteFiles validateFromSnapshot(long snapshotId) {
   protected void validate(TableMetadata base) {
     if (replacedDataFiles.size() > 0) {
       // if there are replaced data files, there cannot be any new row-level deletes for those data files
-      validateNoNewDeletesForDataFiles(base, startingSnapshotId, replacedDataFiles);
+      validateNoNewDeletesForDataFiles(base, startingSnapshotId, null, replacedDataFiles, true);
```

Comment:
It would be nice to have an overload with the original args, since it isn't obvious what `true` is doing here. It's an internal API, but an overload still makes this more readable.

Reply:
Added.